src/main/kotlin/com/exactpro/th2/codec/MessageDatumReader.kt

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2023 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.exactpro.th2.codec
+
+import com.exactpro.th2.common.grpc.Message
+import com.exactpro.th2.common.message.addField
+import com.exactpro.th2.common.value.toValue
+import org.apache.avro.Conversion
+import org.apache.avro.LogicalType
+import org.apache.avro.Schema
+import org.apache.avro.generic.GenericDatumReader
+import org.apache.avro.generic.GenericFixed
+import org.apache.avro.io.Decoder
+import org.apache.avro.io.ResolvingDecoder
+import java.io.IOException
+import java.nio.ByteBuffer
+import javax.xml.bind.DatatypeConverter
+
+class MessageDatumReader(schema: Schema?) :
+    GenericDatumReader<Message.Builder>(schema) {
+
+    @Throws(IOException::class)
+    override fun readWithConversion(
+        old: Any?, expected: Schema?, logicalType: LogicalType?, conversion: Conversion<*>?,
+        decoder: ResolvingDecoder?
+    ): Any? {
+        throw IllegalStateException("Logical types not supported")
```

Comment:
Implement for types described in the spec: https://avro.apache.org/docs/1.10.2/spec.html#Logical+Types
Also, provide extension point for implementing in a child project